docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -23,6 +23,149 @@ not writing data, should return instances of the following services:
+  import io.airlift.units.MinDuration;
+
+  import javax.validation.constraints.NotNull;
+
+  public class ExampleConfig
+  {
+      private String secret;
+      private Duration timeout = Duration.succinctDuration(10, TimeUnit.SECONDS);
+
+      public String getSecret()
+      {
+          return secret;
+      }
+
+      @Config("secret")
+      @ConfigDescription("Secret required to access the data source")
+      @ConfigSecuritySensitive
+      public ExampleConfig setSecret(String secret)
+      {
+          this.secret = secret;
+          return this;
+      }
+
+      @NotNull
+      @MaxDuration("10m")
+      @MinDuration("1ms")
+      public Duration getTimeout()
+      {
+          return timeout;
+      }
+
+      @Config("timeout")
+      public ExampleConfig setTimeout(Duration timeout)
+      {
+          this.timeout = timeout;
+          return this;
+      }
+  }
+
+The example above defines two configuration properties and makes
+the connector more robust by:
+
+* defining all supported properties, which allows detecting typos
+  in the configuration on server startup
+* having a default timeout value, to prevent connections getting stuck
+  indefinitely
+* preventing invalid timeout values (0ms) that would make
+  all requests fail
+* parsing timeout values in different units, detecting invalid values
+* preventing logging the secret value in plain text
+
+The configuration class needs to be bound in a Guice module:
+
+.. code-block:: java
+
+  import com.google.inject.Binder;
+  import com.google.inject.Module;
+
+  import static io.airlift.configuration.ConfigBinder.configBinder;
+
+  public class ExampleModule
+          implements Module
+  {
+      public ExampleModule()
+      {
+      }
+
+      @Override
+      public void configure(Binder binder)
+      {
+          configBinder(binder).bindConfig(ExampleConfig.class);
+      }
+  }
+
+
+And then the module needs to be initialized in the connector factory, when
+creating a new instance of the connector:
+
+.. code-block:: java
+
+  @Override
+  public Connector create(String connectorName, Map<String, String> config, ConnectorContext context)
+  {
+      requireNonNull(config, "config is null");
+      Bootstrap app = new Bootstrap(new ExampleModule());
+      Injector injector = app
+              .doNotInitializeLogging()
+              .setRequiredConfigurationProperties(config)
+              .initialize();
+
+      return injector.getInstance(ExampleConnector.class);
+  }
+
+.. note::
+
+  Environment variables in the catalog properties file
+  (ex. ``secret=${ENV:SECRET}``) will be resolved only when using
+  the ``io.airlift.bootstrap.Bootstrap`` class to initialize the module.
+  See :doc:`/security/secrets` for more information.
+
+If you notice you need to define multiple catalogs using the same connector
+just to change one property, consider adding support for schema and/or
+table properties, that would allow a more fine-grained configuration.
```

Comment:
```
If you end up needing to define multiple catalogs using the same connector
just to change one property, consider adding support for schema and/or
table properties. That would allow a more fine-grained configuration.
```